src/libostree/ostree-repo-static-delta-core.c

```diff
@@ -873,3 +873,70 @@ _ostree_repo_static_delta_dump (OstreeRepo                    *self,
 
   return TRUE;
 }
+
+/**
+ * ostree_repo_static_delta_verify_signature:
+ * @superblock: Static-delta superblock GVariant
+ * @sign: Signature engine used to check superblock
+ * @error: Error
+ *
+ * Verify the superblock data, except inline commits, usin @sign engine.
+ */
+gboolean
+ostree_repo_static_delta_verify_signature (GVariant         *superblock,
+                                           OstreeSign       *sign,
+                                           GError          **error)
+{
+  gsize size = g_variant_type_n_items(G_VARIANT_TYPE (OSTREE_STATIC_DELTA_SUPERBLOCK_FORMAT));
+  g_autoptr(GVariantBuilder) desc_sign_builder = NULL;
+  g_autoptr(GVariant) delta_meta = g_variant_get_child_value (superblock, 0);
+
+  /* Check if there are signatures in the superblock for signature engine */
+  const gchar *signature_key = ostree_sign_metadata_key(sign);
+  GVariantType *signature_format = (GVariantType *) ostree_sign_metadata_format(sign);
+  g_autoptr(GVariant) signatures = g_variant_lookup_value (delta_meta,
+                                                           signature_key,
+                                                           signature_format);
+  if (!signatures)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                   "no signature for '%s' in static-delta superblock",
+                   signature_key);
+      return FALSE;
+    }
+
+  /* Build superblock data for signature check without inline commits and
+   * signatures.
+   */
+  desc_sign_builder = g_variant_builder_new (G_VARIANT_TYPE (OSTREE_STATIC_DELTA_SUPERBLOCK_FORMAT));
+  g_variant_builder_open (desc_sign_builder, G_VARIANT_TYPE ("a{sv}"));
+  g_autoptr(GVariant) metadata = NULL;
+  g_variant_get_child (superblock, 0, "@a{sv}", &metadata);
+  if (metadata != NULL)
+    {
+      GVariantIter iter;
+      GVariant *item;
+
+      g_variant_iter_init (&iter, metadata);
+      while ((item = g_variant_iter_next_value (&iter)))
+        {
+          g_autoptr(GVariant) key_v = g_variant_get_child_value (item, 0);
+          const char *str = g_variant_get_string (key_v, NULL);
+          if (g_str_has_prefix (str, "deltas/") && !g_str_has_suffix (str, "/commitmeta"))
```

Comment:
hm.... so why deltas and commitmeta are not a part of protected data?

Reply:
deltas are already signed, not really useful to signed them again, and commitmeta is part of protected data

Reply:
Is there any chance what these parts are not signed but included?

Reply:
yes, it behaves in the same way then if commits are not inlined
and it uses (very) less memory to create the GVariant to sign/check